### PR TITLE
Apply some range-based for loops

### PR DESCRIPTION
**Pull request by kronbichler**
While browsing through some code, I observed some nasty old-style loops through cells. Change those to range-based for loops.

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]
Looks good to me!

**Review by peterrum — APPROVED** [review]

File: source/dofs/dof_tools.cc [hunk 1]
```diff
@@ -1191,11 +1176,7 @@ namespace DoFTools
     std::vector<types::global_dof_index> dof_indices;
     std::vector<types::global_dof_index> dofs_on_ghosts;
 
-    typename DoFHandler<dim, spacedim>::cell_iterator cell = dof_handler.begin(
-                                                        level),
-                                                      endc =
-                                                        dof_handler.end(level);
-    for (; cell != endc; ++cell)
+    for (const auto &cell : dof_handler.active_cell_iterators())
```
Comment (bangerth) [inline source/dofs/dof_tools.cc:1179]: This is wrong: You're replacing a loop over one level of cells by a loop over all cells. But I think there is a function you can use to get that iterator range.

File: source/dofs/dof_tools.cc [hunk 2]
```diff
@@ -1546,12 +1524,8 @@ namespace DoFTools
...
     else
       {
-        for (typename DoFHandler<dim, spacedim>::active_cell_iterator cell =
-               dof_handler.begin_active();
-             cell != dof_handler.end();
-             cell++)
-          if (cell->is_locally_owned())
-            cell_owners[cell->active_cell_index()] = cell->subdomain_id();
+        for (const auto &cell : dof_handler.active_cell_iterators())
+          cell_owners[cell->active_cell_index()] = cell->subdomain_id();
```
Comment (bangerth) [inline source/dofs/dof_tools.cc:1528]: you accidentally ate the `if (cell->is_locally_owned())`

**Review by bangerth — CHANGES_REQUESTED** [review]
A bunch of MG tests fail, but I think I've pinpointed the two places where you made a mistake.

File: source/dofs/dof_tools.cc (hunk 2 above)
Comment (kronbichler) [reply source/dofs/dof_tools.cc]: Good observation, thanks for this and the other comment. Let's try again.

**Review by bangerth — APPROVED** [review]

**Pull request merged**